src/main/java/pl/orki/hackathon/webapp/user/boundary/graphql/UserMutation.java

```diff
@@ -1,40 +1,55 @@
 package pl.orki.hackathon.webapp.user.boundary.graphql;
 
 import graphql.kickstart.tools.GraphQLMutationResolver;
-import org.jetbrains.annotations.NotNull;
 import org.springframework.stereotype.Component;
+import pl.orki.hackathon.webapp.band.boundary.dto.BandConverter;
+import pl.orki.hackathon.webapp.band.boundary.dto.BandDTO;
+import pl.orki.hackathon.webapp.band.control.BandService;
+import pl.orki.hackathon.webapp.band.entity.Band;
 import pl.orki.hackathon.webapp.user.boundary.UserDTO;
 import pl.orki.hackathon.webapp.user.boundary.UserResponseDTO;
 import pl.orki.hackathon.webapp.user.control.UserService;
-import pl.orki.hackathon.webapp.user.entity.Role;
 import pl.orki.hackathon.webapp.user.entity.User;
+import pl.orki.hackathon.webapp.venue.boundary.dto.VenueConverter;
+import pl.orki.hackathon.webapp.venue.boundary.dto.VenueDTO;
+import pl.orki.hackathon.webapp.venue.control.VenueService;
+import pl.orki.hackathon.webapp.venue.entity.Venue;
 
+import java.util.Optional;
+
+import static pl.orki.hackathon.webapp.user.boundary.UserConverter.convertToEntity;
 import static pl.orki.hackathon.webapp.user.boundary.UserConverter.convertToResponseDTO;
 
 @Component
 public class UserMutation implements GraphQLMutationResolver {
 
     private final UserService userService;
+    private final BandService bandService;
+    private final VenueService venueService;
+    private final BandConverter bandConverter;
+    private final VenueConverter venueConverter;
 
-    public UserMutation(UserService userService) {
+    public UserMutation(UserService userService, BandService bandService, VenueService venueService, BandConverter bandConverter, VenueConverter venueConverter) {
         this.userService = userService;
+        this.bandService = bandService;
+        this.venueService = venueService;
+        this.bandConverter = bandConverter;
+        this.venueConverter = venueConverter;
     }
 
-    public UserResponseDTO createUser(UserDTO userDTO) {
+    public UserResponseDTO createUserBand(UserDTO userDTO, BandDTO bandDTO) {
         User user = convertToEntity(userDTO);
-        userService.createUser(user);
+        Optional<Band> band = bandService.createBand(bandConverter.convertToEntity(bandDTO));
+        band.ifPresent(value -> userService.createUser(user, value.getId()));
```

Comment:
I think that GraphQL query/mutation classes like rest controllers should only handle converting to/from DTO and business logic should remain within services. I would move these two lines to a separate `createUserBand(user, band)` method within `UserService`.

I would also modify `createBand` method so it doesn't return `optional`, because we don't need to query user to create a band. I think it may be even possible to create band together with user this way (if we have User-Band mapping), without persisting band with a separate band:

```
user.setBand(band);
userRepository.save(user);
```